### ISSUE TITLE
Memory Leak with CollectionView

### ISSUE THREAD

**Issue by THoferSchmitz** [dotnet/maui#14993]
### Description

Hi,

I wrote a small tool to download a radio stations playlist for me to look through, very small and simple, and it works. But for some reason for a little over 1k items (less than 3Mb database file size) the app needs more than 1Gb of memory. So there is either a memory leak I don't see, I'm using the CollectionView in a way than I shouldn't or there is a bug somewhere in MAUI / CollectionView

### Steps to Reproduce

My app is a new MAUI project in VS, only the MainPage files were modified, I didn't want to spread this little code over multiple files. Code is in the repo.

On first startup (no db file) it starts at ca. 120Mb, goes up when I use the parse button to 140Mb and the moment the CollectionView loads the files it blows up to more than 1Gb.

### Link to public reproduction project repository

https://github.com/THoferSchmitz/Playlist-Downloader

### Version with bug

6.0.312

### Last version that worked well

6.0.312

### Affected platforms

Windows, I was *not* able test on other platforms

### Affected platform versions

Windows SDK 10.0.17763.0

**Comment by SarthakB26** [comment]
Yes its a memory leak I am also having this issue. It generally happens when items being added/removed from a CollectionView very frequently rather then fixed data source that load only once.
So basically you will see memory increase for each add/remove/update operation in ColletionView ItemSource 

Might be related to
https://github.com/dotnet/maui/issues/14664

**Comment by THoferSchmitz** [comment]
This might be related, depending on the underlying cause. However, I do not replace my ItemSource over and over. This increase (1Gb memory for a 3Mb database) happens on the very first assignment of the property that ItemSource has a binding to. Pressing the parse button again, causing a reassignment of the ItemSource, it will rise to more than 2Gb memory. So I do see that issue as well, however, my initial memory usage is already much bigger than it should be.

**Comment by SarthakB26** [comment]
I would suggest  try ListView with HasUnevenRow="False" if possible Or one of below

https://github.com/roubachof/Sharpnado.CollectionView
https://github.com/Redth/Maui.VirtualListView

**Comment by Dreamescaper** [comment]
VerticalStackLayout does not constraint its content, therefore your CollectionView grows infinitely, rendering all the items at once. 
Use Grid instead of VerticalStackLayout.

**Comment by Dreamescaper** [comment]
Duplicate of https://github.com/dotnet/maui/issues/8097.

**Comment by Dreamescaper** [comment]
Duplicate of #8097.

**Comment by THoferSchmitz** [comment]
I didn't know about the issue with VerticalStackLayout. That helped a little. I don't remember seeing that in the CollectionView docs last time I looked.

However, I would still say it needs to much memory. There are less than 2k items, and they only have a string and an int property and two very small commands. Of course the Collectionview needs the template every time, but even with that the memory usage seems very excessive.

**Comment by mattleibow** [comment]
@jonathanpeppers  would this b better with all the perf work you are doing?

**Comment by jonathanpeppers** [comment]
I think someone should just retest this one w/ main or .NET 8 where we have the fix:

* https://github.com/dotnet/maui/pull/14329

**Comment by Zhanglirong-Winnie** [comment]
Memory leak issue, we verified this issue with Visual Studio Enterprise 17.9.0 Preview 2.1. Can repro on windows platform with sample project.
https://github.com/THoferSchmitz/Playlist-Downloader
![image](https://github.com/dotnet/maui/assets/77434865/bce56bad-be86-49b9-acbd-286ee6d06989)

**Comment by kubaflo** [comment]
Closing as a duplicate of #10007. Same CollectionView memory-leak symptom and report; #10007 is older and covers the broader Android and Windows behavior. Please continue tracking and add any distinct evidence at https://github.com/dotnet/maui/issues/10007.

> Disclosure: this comment was generated by AI.

**Comment by kubaflo** [comment]
Reopening — this issue should not have been closed.

It carries `s/verified` (a maintainer or partner reproduced it), `s/triaged` (it already went through triage). Those labels mean the report was already confirmed as a real product defect, so it is not eligible for backlog cleanup closure. Closure here requires specific fix evidence (a merged PR that closes it, or a `fixed-in-*` label), and there was none.

Sorry for the noise, and thanks to everyone who flagged it.

> Disclosure: this comment was generated by AI.
